src/Adapters/EventSource.Netstandard13/EventSourceTelemetryModule.cs

```diff
@@ -154,9 +154,9 @@ protected override void OnEventSourceCreated(EventSource eventSource)
         private void EnableAsNecessary(EventSource eventSource)
         {
             // Special case: enable TPL activity flow for better tracing of nested activities.
-            if (eventSource.Guid == EventSourceTelemetryModule.TplEventSourceGuid)
+            if (eventSource.Guid == TplActivities.TplEventSourceGuid)
             {
-                this.EnableEvents(eventSource, EventLevel.LogAlways, (EventKeywords)EventSourceTelemetryModule.TaskFlowActivityIds);
+                this.EnableEvents(eventSource, EventLevel.LogAlways, (EventKeywords) TplActivities.TaskFlowActivityIdsKeyword);
```

Comment:
EventSource works with Hierarchical IDs.     There was a problem where it only worked with ETW and not EventListeners (as used here), however that has been fixed in .NET Core (bug still pending to port it to Desktop).   It does work on Core